src/drivers/esp8266_drv.c

```diff
@@ -31,82 +31,150 @@
 #include "queue.h"
 #include "str_util.h"
 #include "task.h"
+#include "taskUtil.h"
 #include "wifi.h"
 #include <stdbool.h>
 #include <stdlib.h>
 #include <string.h>
 
-#define _AT_TASK_TIMEOUT_MS	3
-#define _BAD_STATE_BACKOFF_MS	3000
-#define _CLIENT_BACKOFF_MS	30000
-#define _INIT_FAIL_SLEEP_MS	10000
-#define _LOG_PFX		"[ESP8266 Driver] "
-#define _MAX_CHANNELS		5
-#define _SERIAL_BAUD		115200
-#define _SERIAL_BITS		8
-#define _SERIAL_CMD_MAX_LEN	1024
-#define _SERIAL_PARITY		0
-#define _SERIAL_RX_BUFF_SIZE	256
-#define _SERIAL_STOP_BITS	1
-#define _SERIAL_TX_BUFF_SIZE	256
-#define _TASK_STACK_SIZE	512
-#define _TASK_THREAD_NAME	"ESP8266 Driver"
-
-enum _cmd {
-        _CMD_UNKNOWN = 0,
-        _CMD_NOOP,
-        _CMD_INIT,
-        _CMD_CLIENT_AP_GET,
-        _CMD_CLIENT_AP_SET,
-        _CMD_CLIENT_IP_GET,
-        _CMD_DAEMON_SETUP,
+#define AT_TASK_TIMEOUT_MS	3
+#define CHECK_DONE_SLEEP_MS	3600000
+#define CLIENT_BACKOFF_MS	30000
+#define INIT_FAIL_SLEEP_MS	10000
+#define LOG_PFX			"[ESP8266 Driver] "
+#define MAX_CHANNELS		5
+#define SERIAL_BAUD		115200
+#define SERIAL_BITS		8
+#define SERIAL_CMD_MAX_LEN	1024
+#define SERIAL_PARITY		0
+#define SERIAL_RX_BUFF_SIZE	256
+#define SERIAL_STOP_BITS	1
+#define SERIAL_TX_BUFF_SIZE	256
+#define TASK_STACK_SIZE		512
+#define TASK_THREAD_NAME	"ESP8266 Driver"
+
+struct device {
+        struct Serial *serial;
+        enum dev_init_state init_state;
 };
 
-struct _client {
-        struct esp8266_client_info info;
+struct client {
         const struct wifi_client_cfg *config;
-        tiny_millis_t next_check_ms;
+        struct esp8266_client_info info;
+        tiny_millis_t next_join_attempt;
 };
 
-struct _daemon {
+struct daemon {
```

Comment:
I know the 'daemon' naming was introduced previously, but want to suggest it could be named better. Is it truly a background process as the name suggests? https://en.wikipedia.org/wiki/Daemon_(computing)

If you agree we can file a separate issue.